k8sclient/k8sclient.go

```diff
@@ -99,8 +100,21 @@ func parsePodNetworkObject(podnetwork string) ([]map[string]interface{}, error)
 
 	// Parse the podnetwork string, and assume it is JSON.
 	if err := json.Unmarshal([]byte(podnetwork), &podNet); err != nil {
-		// If the JSON parsing fails, assume it is comma delimited.
+
+		// Check and see if it matches the specification for the comma delimited format.
+		// "Valid attachment names must be comprised of units of the DNS-1123 label format"
+		// [a-z0-9]([-a-z0-9]*[a-z0-9])?
+		// And we allow at (@), comma (,), and forward slash (/)
+		// But it must start and end alphanumerically.
+		matched, _ := regexp.MatchString("^[a-z0-9]([-a-z0-9/@,]*[a-z0-9])?$", podnetwork)
```

Comment:
podnetwork := "netA,netB,example-namespace/attachment-name" is valid network annotation. Please correct me, if I am miss understood concept here.

https://play.golang.org/p/tllaAF8S8K- just started testing it

Reply:
We also need to take care of space ' ', like "netA, netB". After here, parsePodNetworkObjectName() trims spaces, at that time, we don't do that. So regexp should take care of space as well as '@,/', otherwise user cannot use space between network name.

Reply:
In addition, should [-a-z] be [a-z]?
There is no notation [-a-z] in https://github.com/google/re2/wiki/Syntax

Reply:
@s1061123 regarding `[-a-z]` I based it on [this validation in kube](https://github.com/kubernetes/kubernetes/blob/8d7d7a5e0d4d7e75f5a860574346944b8cc0fc43/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L107) -- I believe that (at least in PCRE) you can put a `-` as the first or last item in a character class in order to have it be a literal dash, instead of being a range, or you can escape it.

Good call regarding the spaces -- next change I'm submitting will have it split by commas, and then I'm going to let your routine for parsePodNetworkObjectName and then validate the items therein as I realized that it became tricky to validate good looking names, e.g. this regex would've also accepted bad input like `foo@@@@bar////quux`, too.